player/lua/osc.lua

```diff
@@ -1890,10 +1890,11 @@ function osc_init()
     ne.content = "\238\132\132"
     ne.eventresponder["mbtn_left_up"] =
         function ()
-            mp.commandv("add", "chapter", -1)
-            if user_opts.chapters_osd then
-                show_message(get_chapterlist(), 3)
-            end
+            mp.command_native_async({"add", "chapter", -1}, function()
```

Comment:
are you sure `add` actually provides async feedback? it's not mentioned in the manual

Reply:
The manual says it's like `mp.command_native`, except that a function is called upon completion.

And `mp.command_native` is like `mp.commandv`, except that the argument list is passed as a table.

Reply:
Sadly, this doesn't fix it for me. I can still reproduce the problem. I don't know if it matters but the sample I used happened to have ordered chapters although I could see it occur in the non-ordered chapters part.

Reply:
I've done a bit of digging and I think I have a better understanding of the problem.

the `add` command, as @sfan5  pointed out, does not provide async feedback

However, the chapter property determines what chapter it is based on what the current time position is, and seeking takes place in mpv's playloop.

I think it worked for me because when we make it an async call, the command instead gets executed when the function `mp_process_input` get called, which is also in the playloop, and by that time, `execute_queued_seek` has already been called.

Reply:
Maybe the better solution would be to just artificially determine what the new chapter is supposed to be, rather waiting for seeking to take place so that the chapter property returns the correct value.

Reply:
Thinking about this some more, I don't think is specifically an osc issue. The OSD can display the wrong chapter as well if you go back and forth with the keybinds. Something about `show_property_osd` I think.

Reply:
In the function `get_current_chapter`, there's this line:
```c
    return MPMAX(mpctx->last_chapter_seek, i - 1);
```
so when we're skipping to the next chapter, `last_chapter_seek` is the larger number, which is why we pretty much never see inconsistencies when skipping forward, only skipping back when `last_chapter_seek` would be the smaller number